src/mono/msbuild/apple/build/AppleBuild.LocalBuild.props

```diff
@@ -84,4 +83,7 @@
     <MonoTargetsTasksAssemblyPath>$([MSBuild]::NormalizePath('$(MonoTargetsTasksDir)', 'MonoTargetsTasks.dll'))</MonoTargetsTasksAssemblyPath>
   </PropertyGroup>
 
+  <!-- Due to dependencies on resolving properties like: $(MicrosoftNetCoreAppRuntimePackRidDir) we need to import the default props at the end -->
+  <Import Project="$(MSBuildThisFileDirectory)AppleBuild.props" />
```

Comment:
We should probably consider inverting the logic and instead import `AppleBuild.LocalBuild.props` from `AppleBuild.props` since LocalBuild.props resolve a lot of used properties throughout the build like:
- MicrosoftNetCoreAppRuntimePackDir
- MicrosoftNetCoreAppRuntimePackRidDir
- Etc

These properties are then again reevaluated in `AppleBuild.targets`:
https://github.com/dotnet/runtime/pull/91800/files#diff-e351a598f31c0a001f31622d9babc297dbcbe5827eabf55ab92700e1839e70f1R46-R49
which wont be needed anymore if we invert the logic.

Proposal:
- Rename: `AppleBuild.LocalBuild.props` to `AppleBuild.Helix.props` 
- Add something like:
```xml
<Import Condition="'$(IsHelixBuild)' == 'true'" Project="AppleBuild.Helix.props" />
```
in the `AppleBuild.props`

This would be probably a lot cleaner and could reduce the amount of properties we are resolving in the `*.targets` files

Reply:
Sounds good to me. I suggest to have a single entry point, like `AppleBuild.props` and to add logic for importing Helix targets `<Import Condition="'$(IsHelixBuild)' == 'true'" Project="AppleBuild.Helix.props" />` under the hood.

Reply:
That is exactly what I suggested above :)